libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt

```diff
@@ -1,21 +1,25 @@
 package com.mapbox.navigation.core.replay.history
 
-import android.util.Log
 import com.google.gson.Gson
 import com.google.gson.internal.LinkedTreeMap
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
 
 typealias CustomEventMapper = (String, LinkedTreeMap<*, *>) -> ReplayEventBase?
 
 /**
  * This class is responsible for creating [ReplayEvents] from history data.
  *
- * @param gson Gson
+ * @param gson Gson (optional)
  * @param customEventMapper if you have added custom events to the replay history, include your
  * own [CustomEventMapper] (optional)
+ * @param logger interface for logging any events (optional)
  */
 class ReplayHistoryMapper(
     private val gson: Gson = Gson(),
-    private val customEventMapper: CustomEventMapper? = null
+    private val customEventMapper: CustomEventMapper? = null,
+    private val logger: Logger? = null
```

Comment:
Same as in https://github.com/mapbox/mapbox-navigation-android/pull/2774#discussion_r412323388 right @kmadsen?